packages/@aws-cdk/cloudformation-diff/lib/diff-template.ts

```diff
@@ -53,10 +54,16 @@ export function fullDiff(
 
   normalize(currentTemplate);
   normalize(newTemplate);
-  const theDiff = diffTemplate(currentTemplate, newTemplate);
+  let theDiff = diffTemplate(currentTemplate, newTemplate);
   if (changeSet) {
-    filterFalsePositives(theDiff, changeSet);
-    addImportInformation(theDiff, changeSet);
+    // These methods mutate the state of theDiff, using the changeSet.
+    const changeSetDiff = new TemplateAndChangeSetDiffMerger({ changeSet: changeSet });
```

Comment:
Nit: You can consolidate key-value pairs with the same name down to a single value. For example, this could just be written as:
```ts
const changeSetDiff = new TemplateAndChangeSetDiffMerger({ changeSet });
```